### PR TITLE
The bug related to KNearest CentroidUpdateOn AxialInterpolation fixed

### PR DESCRIPTION
**Pull request by wuwenbin2006**
Now it is possible to turn off the knearest option just by cmfd.setCentroidUpdateOn(false).

### CODE REVIEW

File: profile/Makefile [hunk 1]
```diff
@@ -2,10 +2,10 @@
 # User Options
 #===============================================================================
 
-COMPILER    = gnu
+COMPILER    = mpi
 OPENMP      = yes
-OPTIMIZE    = yes
-DEBUG       = no
+OPTIMIZE    = no
+DEBUG       = yes
```
Comment (GiudGiud) [inline profile/Makefile]: These changes don't really need to part of this PR

File: profile/models/simple-lattice/simple-lattice-3d.cpp [hunk 2]
```diff
@@ -22,7 +22,7 @@ int main(int argc, char* argv[]) {
   double polar_spacing = 0.5;
   int num_polar = 2;
   double tolerance = 1e-5;
-  int max_iters = 1000;
+  int max_iters = 3;
```
Comment (GiudGiud) [inline profile/models/simple-lattice/simple-lattice-3d.cpp]: No need to change this here

File: profile/models/simple-lattice/simple-lattice-3d.cpp [hunk 3]
```diff
@@ -212,16 +212,17 @@ int main(int argc, char* argv[]) {
   for (int g=3; g<7; g++)
     cmfd_group_structure.at(1).push_back(g+1);
   cmfd.setGroupStructure(cmfd_group_structure);
-  cmfd.setKNearest(1);
+  cmfd.setKNearest(9);
   cmfd.useFluxLimiting(false);
-
+  cmfd.setCentroidUpdateOn(false);
+  cmfd.useAxialInterpolation(true);
```
Comment (GiudGiud) [inline profile/models/simple-lattice/simple-lattice-3d.cpp]: Could use a small comment for these lines, not absolutely necessary though

File: src/Cmfd.cpp [hunk 4]
```diff
@@ -237,7 +237,7 @@ Cmfd::~Cmfd() {
       delete [] _old_boundary_flux;
       delete [] _boundary_surface_currents;
     }
-    delete _domain_communicator;
+    //delete _domain_communicator;
```
Comment (GiudGiud) [inline src/Cmfd.cpp]: domain communicator is already deleted line 223

File: src/Cmfd.cpp [hunk 5]
```diff
@@ -2367,91 +2367,91 @@ void Cmfd::setQuadrature(Quadrature* quadrature) {
...
-      _axial_interpolants.at(r)[j] = 0.0;
-  }
 
   /* Compute axial quadratic interpolation values if requested */
   if (_use_axial_interpolation && _local_num_z >= 3) {
 
+    /* Initialize axial quadratic interpolant values */
+    _axial_interpolants.resize(_num_FSRs);
+    for (long r=0; r < _num_FSRs; r++) {
+      _axial_interpolants.at(r) = new double[3];
```
Comment (GiudGiud) [inline src/Cmfd.cpp]: Initializing with new double[3]\() eliminates the need to set each element to 0

File: src/Cmfd.cpp [hunk 6]
```diff
@@ -2637,11 +2637,10 @@ CMFD_PRECISION Cmfd::getFluxRatio(int cell_id, int group, int fsr) {
...
   double* interpolants;
   double ratio = 1.0;
-  if (_k_nearest > 0)
+  if (_use_axial_interpolation)
     interpolants = _axial_interpolants.at(fsr);
-
-  if (_k_nearest > 0 and (interpolants[0] != 0 or interpolants[2] != 0)) {
-
+  if (_use_axial_interpolation && _local_num_z >= 3 && 
+    ( interpolants[0] != 0 || interpolants[2] != 0) ) {
```
Comment (GiudGiud) [inline src/Cmfd.cpp]: (interpolants[0] != 0 || interpolants[2] != 0))

File: src/Cmfd.cpp (hunk 4 above)
Comment (GiudGiud) [reply src/Cmfd.cpp]: (remove the line)

**Comment by GiudGiud** [comment]
@wuwenbin2006 Do not merge your own PRs. Especially after a review, since I have to check that what you did addressed the review, then I'll merge

Please make sure to address these 3 comments in your next PR

File: src/Cmfd.cpp (hunk 6 above)
Comment (wuwenbin2006) [reply src/Cmfd.cpp]: Do you mean avoiding comparing the equality of float?

Comment (GiudGiud) [reply src/Cmfd.cpp]: No it's a style comment. Parenthesis don't need an extra whitespace inside

**Pull request merged**